inference-engine/src/gna_plugin/backend/dnn_components.cpp

```diff
@@ -8,32 +8,62 @@
 #include <ie_common.h>
 #include <legacy/ie_layers.h>
 #include <iomanip>
+#include <details/caseless.hpp>
+#include <layers/gna_copy_layer.hpp>
 #include "backend/dnn_types.h"
 
 #include "dnn_components.hpp"
 
 using namespace GNAPluginNS;
+using namespace GNAPluginNS::backend;
 
-intel_dnn_component_t & backend::DnnComponents::addComponent(const std::string layerName, const std::string layerMetaType) {
-    components.emplace_back(layerName, intel_dnn_component_t());
-    auto &currentComponent = components.back().second;
+intel_dnn_component_t & DnnComponents::addComponent(const std::string layerName, const std::string layerMetaType) {
+    auto isDelayed = InferenceEngine::details::CaselessEq<std::string>()(layerMetaType, DelayedCopyLayerName);
+    delayedOperations += isDelayed ? 1 : 0;
+    components.emplace_back(DnnComponentExtra{layerName, {}, isDelayed});
+    auto &currentComponent = components.back().dnnComponent;
 #ifdef PLOT
-    currentComponent.original_layer_name = components.back().first.c_str();
+    currentComponent.original_layer_name = components.back().name.c_str();
     std::cout << "IR layer : " << std::left << std::setw(20) << layerName << " " << layerMetaType << "_" << components.size() - 1 << std::endl;
 #endif
+    int execOrder = 0;
+    if (!isDelayed) {
+        execOrder = static_cast<int>(components.size() - 1 - delayedOperations);
+    } else {
+        // todo: not perfect - propose to create mapping table that will be printed out by extra request
+        execOrder = - static_cast<int>(delayedOperations);
```

Comment:
I still don't understand why we return a negative number here.

Reply:
Please reach out to @esmirno, since he is the original code creator. I believe that it will allow to easily see which layer is delayed and which is not, while keeping their order and separating delayed and non-delayed operations (for eg you know that that execOrder 1 is 2nd non-delayed op while execOrder = -4 means 4th delayed op)